pkg/placement/placement.go

```diff
@@ -222,33 +216,31 @@ func (p *Service) ProcessHost(host *placementv1pb.Host) {
 	updateRequired := false
 
 	for _, e := range host.Entities {
-		func() {
-			p.entriesLock.Lock()
-			defer p.entriesLock.Unlock()
-			if _, ok := p.entries[e]; !ok {
-				p.entries[e] = NewConsistentHash()
-			}
+		p.entriesLock.Lock()
 
-			exists := p.entries[e].Add(host.Name, host.Id, host.Port)
-			if !exists {
-				updateRequired = true
-				monitoring.RecordPerActorTypeReplicasCount(e, host.Name)
-			}
-		}()
+		if _, ok := p.entries[e]; !ok {
+			p.entries[e] = NewConsistentHash()
+		}
+
+		exists := p.entries[e].Add(host.Name, host.Id, host.Port)
+		if !exists {
+			updateRequired = true
+			monitoring.RecordPerActorTypeReplicasCount(e, host.Name)
+		}
+
+		p.entriesLock.Unlock()
```

Comment:
Not a review comment, but a question for my own learning. Does golang have a scoped lock? I know about defer to the end of the function execution, but what about to the end of the scope?

Reply:
Personally, I do not prefer using `defer`. To implement the scoped lock, as you see the original code, local func and defer is one of the options.

```
func() {
   lock
  defer unlock
  ...
}
```

`func() {}` is the scope and defer will execute at the end of this scope.